diff --git a/vm_tools/concierge/vm_util.cc b/vm_tools/concierge/vm_util.cc
index 8adb62c13262..78ebda00dcbb 100644
--- a/vm_tools/concierge/vm_util.cc
+++ b/vm_tools/concierge/vm_util.cc
@@ -228,6 +228,9 @@ Disk::~Disk() = default;
 int64_t GetVmMemoryMiB() {
   int64_t sys_memory_mb = base::SysInfo::AmountOfPhysicalMemoryMB();
   int64_t vm_memory_mb;
+
+  LOG(INFO) << "AmountOfPhysicalMemoryMB: " << sys_memory_mb << "MB";
+
   if (sys_memory_mb >= 4096) {
     // On devices with <=4GB RAM, reserve 1GB for other processes.
     vm_memory_mb = sys_memory_mb - 1024;
@@ -235,6 +238,8 @@ int64_t GetVmMemoryMiB() {
     vm_memory_mb = sys_memory_mb / 4 * 3;
   }
 
+  LOG(INFO) << "vm_memory_mb: " << vm_memory_mb << "MB";
+
   // Limit guest memory size to avoid running out of host process address space.
   //
   // A 32-bit process has 4GB address space, and some parts are not usable for
