diff --git a/concierge/vm_util.cc b/concierge/vm_util.cc
index 8adb62c132..c11ae9eb60 100644
--- a/concierge/vm_util.cc
+++ b/concierge/vm_util.cc
@@ -230,9 +230,9 @@ int64_t GetVmMemoryMiB() {
   int64_t vm_memory_mb;
   if (sys_memory_mb <= 4096) {
     // On devices with <=4GB RAM, reserve 1GB for other processes.
-    vm_memory_mb = sys_memory_mb - 1024;
+    vm_memory_mb = sys_memory_mb - 2048;
   } else {
-    vm_memory_mb = sys_memory_mb / 4 * 3;
+    vm_memory_mb = sys_memory_mb / 3 * 2;
   }
 
   // Limit guest memory size to avoid running out of host process address space.
